pkg/manifests/manifests.go

```diff
@@ -843,9 +858,72 @@ func (f *Factory) PrometheusK8s(host string) (*monv1.Prometheus, error) {
 		}
 	}
 
+	if trustedCABundleCM != nil {
+		volumeName := "prometheus-trusted-ca-bundle"
+		p.Spec.Containers = append(p.Spec.Containers, v1.Container{
+			Name: "prometheus",
+			VolumeMounts: []v1.VolumeMount{
+				trustedCABundleVolumeMount(volumeName, PrometheusTrustedCABundleDir),
+			},
+		})
+		p.Spec.Volumes = append(p.Spec.Volumes, trustedCABundleVolume(trustedCABundleCM.Name, volumeName))
+	}
+
+	telemetryEnabled := f.config.TelemeterClientConfig.IsEnabled()
+	if telemetryEnabled {
+
+		selectorRelabelConfig, err := relabelgen.LabelSelectorsToRelabelConfig(f.config.PrometheusK8sConfig.TelemetryMatches)
+		if err != nil {
+			return nil, errors.Wrap(err, "generate label selector relabel config")
+		}
+
+		compositeToken, err := json.Marshal(map[string]string{
+			"cluster_id":          f.config.TelemeterClientConfig.ClusterID,
+			"authorization_token": f.config.TelemeterClientConfig.Token,
+		})
+
+		spec := monv1.RemoteWriteSpec{
+			URL:         f.config.TelemeterClientConfig.TelemeterServerURL,
+			BearerToken: base64.StdEncoding.EncodeToString(compositeToken),
+			WriteRelabelConfigs: []monv1.RelabelConfig{
+				*selectorRelabelConfig,
+				monv1.RelabelConfig{
+					TargetLabel: "_id",
+					Replacement: f.config.TelemeterClientConfig.ClusterID,
+				},
+				monv1.RelabelConfig{
```

Comment:
Why is this needed?

Reply:
The main reason is to be able to distinguish between alerts produced in cluster (`alerts`) and alerts produced on the receiving side (`ALERTS`). The naming is a little unfortunate I believe, but we need to stick to this for compatibility for older data.

Reply:
let's add the above statement as a code comment.

Reply:
added